### ISSUE TITLE
Make runtest failed

### ISSUE THREAD

**Issue by SaddamRaut** [tidsp/caffe-jacinto#19]
Caffe jacinto build failed. Details are given below-

Make runtest

Please suggest solution.

GPU : Tesla K20c

Error:

make runtest 
.build_release/tools/caffe
I0605 16:20:30.894817  3694 caffe.cpp:807] This is NVCaffe 0.16.4 started at Tue Jun  5 16:20:30 2018
I0605 16:20:30.895033  3694 caffe.cpp:810] CuDNN version: 6021
I0605 16:20:30.895043  3694 caffe.cpp:811] CuBLAS version: 8000
I0605 16:20:30.895050  3694 caffe.cpp:812] CUDA version: 8000
I0605 16:20:30.895058  3694 caffe.cpp:813] CUDA driver version: 9000
caffe: command line brew
usage: caffe <command> <args>

commands:
  train           train or finetune a model
  test            score a model
  device_query    show GPU diagnostic information
  time            benchmark model execution time

  Flags from tools/caffe.cpp:
    -ap_version (Average Precision type for object detection) type: string
      default: "11point"
    -display_sparsity (Display the amount of sparsity) type: bool
      default: false
    -gpu (Optional; run in GPU mode on given device IDs separated by ', '.Use
      '-gpu all' to run on all available GPUs. The effective training batch
      size is multiplied by the number of devices.) type: string default: ""
    -iterations (The number of iterations to run.) type: int32 default: 50
    -model (The model definition protocol buffer text file.) type: string
      default: ""
    -output (Prefix for output prototxt and caffemodel) type: string
      default: ""
    -show_per_class_result (Show per class result for object detection)
      type: bool default: true
    -sighup_effect (Optional; action to take when a SIGHUP signal is received:
      snapshot, stop or none.) type: string default: "snapshot"
    -sigint_effect (Optional; action to take when a SIGINT signal is received:
      snapshot, stop or none.) type: string default: "stop"
    -snapshot (Optional; the snapshot solver state to resume training.)
      type: string default: ""
    -solver (The solver definition protocol buffer text file.) type: string
      default: ""
    -threshold_fraction_high (Optional: fraction of weights to be zeroed)
      type: double default: 0.90000000000000002
    -threshold_fraction_low (Optional: fraction of weights to be zeroed)
      type: double default: 0.40000000000000002
    -threshold_fraction_mid (Optional: fraction of weights to be zeroed)
      type: double default: 0.80000000000000004
    -threshold_step_factor (Optional: step factor to find the optimal threshold
      for each layer) type: double default: 1e-08
    -threshold_value_max (Optional: vlues of weights below which are zeroed)
      type: double default: 0.0001
    -threshold_value_maxratio (Optional: determine threshold as a factor of
      max) type: double default: 0.001
    -weights (Optional; the pretrained weights to initialize finetuning,
      separated by ', '. Cannot be set simultaneously with snapshot.)
      type: string default: ""
.build_release/test/test_all.testbin 0 --gtest_shuffle 
Cuda number of devices: 1
Setting to use device 0
Current device id: 0
Current device name: Tesla K20c
Note: Randomizing tests' orders with a seed of 77198 .
[==========] Running 1906 tests from 266 test cases.
[----------] Global test environment set-up.
[----------] 4 tests from SoftmaxWithLossLayerTest/3, where TypeParam = caffe::GPUDevice<double>
[ RUN      ] SoftmaxWithLossLayerTest/3.TestGradient
F0605 16:20:32.456480  3715 math_functions.cu:508] Check failed: error == cudaSuccess (8 vs. 0)  invalid device function
*** Check failure stack trace: ***
    @     0x7fe4fff8c5cd  google::LogMessage::Fail()
    @     0x7fe4fff8e433  google::LogMessage::SendToLog()
    @     0x7fe4fff8c15b  google::LogMessage::Flush()
    @     0x7fe4fff8ee1e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fe4fdef85d2  caffe::caffe_gpu_set<>()
    @     0x7fe4fda95c9d  caffe::Tensor::gpu_set()
    @     0x7fe4fda95eb0  caffe::Tensor::set()
    @           0x4bea78  caffe::Layer<>::SetLossWeights()
    @     0x7fe4fde45249  caffe::SoftmaxWithLossLayer<>::LayerSetUp()
    @           0x4bb0a5  caffe::LayerBase::SetUp()
    @           0x4df53f  caffe::GradientChecker<>::CheckGradientExhaustive()
    @           0x5f5ccc  caffe::SoftmaxWithLossLayerTest_TestGradient_Test<>::TestBody()
    @           0x86e3a3  testing::internal::HandleExceptionsInMethodIfSupported<>()
    @           0x867d1a  testing::Test::Run()
    @           0x867e68  testing::TestInfo::Run()
    @           0x867f45  testing::TestCase::Run()
    @           0x8681ff  testing::internal::UnitTestImpl::RunAllTests()
    @           0x868533  testing::UnitTest::Run()
    @           0x46b909  main
    @     0x7fe4fcb2f830  __libc_start_main
    @           0x475329  _start
    @              (nil)  (unknown)
Makefile:565: recipe for target 'runtest' failed
make: *** [runtest] Aborted (core dumped)

**Comment by mathmanu** [comment]
can you try:
make -j16

and if it finishes successfully, please try the following as well:
make pycaffe

**Comment by mathmanu** [comment]
We have now published caffe-0.17 branch. Can you please try building it?

**Comment by SaddamRaut** [comment]
this issue is not found in  caffe-0.17
